app/lib/shared/configuration.dart

```diff
@@ -66,17 +66,18 @@ class Configuration {
         popularityDumpBucketName = PackagePopularity.bucketName(false),
         searchSnapshotBucketName = 'dartlang-pub--search-snapshot';
 
-  /// Create a configuration for development/staging deployment.
-  Configuration._dev()
-      : projectId = 'dartlang-pub-dev',
-        packageBucketName = 'dartlang-pub-dev--pub-packages',
-        analyzerServicePrefix =
-            'https://analyzer-dot-dartlang-pub-dev.appspot.com',
-        dartdocServicePrefix =
-            'https://dartdoc-dot-dartlang-pub-dev.appspot.com',
-        searchServicePrefix = 'https://search-dot-dartlang-pub-dev.appspot.com',
+  /// Base configuration for local development
+  Configuration._local(String projectId)
+      : projectId = projectId,
+        packageBucketName = '$projectId--pub-packages',
+        analyzerServicePrefix = 'https://analyzer-dot-$projectId.appspot.com',
+        dartdocServicePrefix = 'https://dartdoc-dot-$projectId.appspot.com',
+        searchServicePrefix = 'https://search-dot-$projectId.appspot.com',
         popularityDumpBucketName = PackagePopularity.bucketName(true),
```

Comment:
This is in pkg/_popularity/lib/popularity.dart, and if you change that too, both the _dev() and the _prod() constructor can become just a call to _local(<project-id>)